tachyon/math/base/semigroups.h

```diff
@@ -301,6 +301,23 @@ class AdditiveSemigroup {
     }
   }
 
+  // generator: g
+  // return: [c, c * g, c * 2 * g, ..., c * (|size| - 1) * g]
+  constexpr static std::vector<ReturnTy> GetSuccessiveScalarMuls(
+      size_t size, const G& generator, const G& c = G::One()) {
```

Comment:
Does c here also have something to do with the coset of the evaluataion domain? I'm not sure if we need `c` for the `AdditiveSemigroup` case

Reply:
That would be coset or tau! and it reminds me that I should apply this when setting up kzg parameters.

Reply:
This is because it is a method of `AdditiveSemigroup` which corresponds to `GetSuccessivePowers()` of `MultiplicativeSemigroup`. As you said, there's no usecase.